sw/device/tests/autogen/plic_all_irqs_test.c

```diff
@@ -0,0 +1,211 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_uart.h"
+#include "sw/device/lib/handler.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+static dif_uart_t uart0;
+static dif_rv_plic_t plic;
+
+static const top_earlgrey_plic_target_t kHart = kTopEarlgreyPlicTargetIbex0;
+
+/**
+ * Flag indicating which peripheral is under test.
+ *
+ * Declared volatile because it is referenced in the main program flow as well
+ * as the ISR.
+ */
+static volatile top_earlgrey_plic_peripheral_t peripheral_expected;
+
+/**
+ * Flags indicating the irq expected to have triggered and serviced within the
+ * peripheral.
+ *
+ * This is the index of the interrupt in the peripheral's INTR_STATE register.
+ * Declared volatile because it is referenced in the main program flow as well
+ * as the ISR.
+ */
+static volatile uint32_t irq_expected;
+static volatile uint32_t irq_serviced;
```

Comment:
Is there a reason these can't be dif_##_peripheral##_irq_t types throughout? (to avoid casting)

Reply:
It can be, but it will end up getting expanded to 70 variables (2 per IP). That's probably ok though. Updated.